### PR TITLE
chore: merge queue

### LINKED ISSUE #486: Merge Queues in CI to speed up dependabot updates
### Describe what should be investigated or refactored

Like [zarf](https://github.com/zarf-dev/zarf/tree/main/.github/workflows), we should adopt merge queues in Pepr.

A merge queue is a tool or process designed to manage and streamline the integration of pull requests (PRs) into a project's main branch. When a pull request is approved and ready to merge, it wil enter the merge queue instead of being merged immediately. The queue processes each PR sequentially, performing checks before integration.

After any previously queued PR is merged, the next PR is rebased or synchronized with the updated main branch and then retested. This guarantees that the pull request is validated against the most recent state of the codebase.

### Links to any relevant code

Reference the `.github` workflow files in Pepr.

### Additional context

Github docs on [merge queues](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-a-merge-queue).

### PR DESCRIPTION
**Pull request by cmwylie19**
## Description

Add merge queues to KFC

## Related Issue

Fixes #486 

Relates to #

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor Guide Steps](https://docs.pepr.dev/main/contribute/#submitting-a-pull-request) followed

### CODE REVIEW

File: .github/workflows/dependency-review.yml [hunk 1]
```diff
@@ -7,8 +7,17 @@
...
 # Source repository: https://github.com/actions/dependency-review-action
 name: Dependency Review
-on: [pull_request]
-
+on: 
+  pull_request:
+    paths-ignore:
+    - "LICENSE"
+    - "CODEOWNERS"
+    - "**.md"
```
Comment (cmwylie19) [inline .github/workflows/dependency-review.yml:15]:
```suggestion
```

Stick to the scope of the story

File: .github/workflows/commitlint.yml [hunk 2]
```diff
@@ -4,6 +4,11 @@ on:
   pull_request:
     branches: [main]
     types: [opened, edited, synchronize]
+  merge_group:
+    paths-ignore:
+      - "LICENSE"
+      - "CODEOWNERS"
+      - "**.md"
```
Comment (cmwylie19) [inline .github/workflows/commitlint.yml:11]:
```suggestion
```

File: .github/workflows/codeql.yml [hunk 3]
```diff
@@ -13,6 +13,11 @@ name: CodeQL
 
 on:
   push:
+  merge_group:
+    paths-ignore:
+      - "LICENSE"
+      - "CODEOWNERS"
+      - "**.md"
```
Comment (cmwylie19) [inline .github/workflows/codeql.yml:20]:
```suggestion
```

**Review by btlghrants — APPROVED** [review]
Yup.

**Review by btlghrants — DISMISSED** [review]
Yep

**Review by btlghrants — DISMISSED** [review]

**Review by btlghrants — APPROVED** [review]

File: .github/workflows/node.js.yml [hunk 4]
```diff
@@ -27,6 +28,7 @@ jobs:
       - name: Verify the integrity of provenance attestations and registry signatures for installed dependencies
         run: npm audit signatures
   test:
+    if: ${{ github.event.pull_request && github.event.pull_request.title }}
```
Comment (btlghrants) [inline .github/workflows/node.js.yml:31]:
```suggestion
    if: ${{ github.event_name == 'pull_request' }}
```

**Review by btlghrants — APPROVED** [review]

**Review by btlghrants — COMMENTED** [review]
Worth a try. :man_shrugging:

**Review by btlghrants — DISMISSED** [review]

**Review by btlghrants — APPROVED** [review]

**Review by btlghrants — APPROVED** [review]

**Review by btlghrants — APPROVED** [review]

**Pull request merged**